Python/compile.c

```diff
@@ -4564,7 +4566,9 @@ compiler_async_with(struct compiler *c, stmt_ty s, int pos)
     withitem_ty item = asdl_seq_GET(s->v.AsyncWith.items, pos);
 
     assert(s->kind == AsyncWith_kind);
-    if (c->u->u_scope_type != COMPILER_SCOPE_ASYNC_FUNCTION) {
+    if (c->c_flags->cf_flags & PyCF_ALLOW_TOP_LEVEL_AWAIT){
+        c->u->u_ste->ste_coroutine = 1;
```

Comment:
Have you checked if this `c->u->u_ste->ste_coroutine = 1;` line is needed?  Does it work without it?

Reply:
yes it is necessary, otherwise: 

```
>>> async with acm() as delta:
>>>     count += delta

Traceback (most recent call last):
  File "minirepl.py", line 97, in f
    co = compile(inp, '', 'single', flags=PyCF_ALLOW_TOP_LEVEL_AWAIT)
  File "<string>", line 2
SyntaxError: 'async with' outside async function
```

Reply:
OK, cool, let's keep it then :)

Reply:
And there is be a test in `Lib/test/test_builtin.py` for that  in case you ask.